sparkmagic/sparkmagic/utils/configuration.py

```diff
@@ -48,15 +53,17 @@ def get_livy_kind(language):
 def get_auth_value(username, password):
     if username == '' and password == '':
         return constants.NO_AUTH
-    
+
     return constants.AUTH_BASIC
 
 
 # Configs
 
- 
+
 def get_session_properties(language):
-    properties = copy.deepcopy(session_configs())
+    config = {'session_configs': copy.deepcopy(session_configs())}
+    _merge_required(config, path)
+    properties = config['session_configs']
```

Comment:
If `session_configs()` is called elsewhere in sparkmagic, will the required fields merged into it?

Reply:
this covers all session_configs paths that create new sessions. however, i think it does make sense to move this merge logic directly into session_configs. the only thing i'm a little concerned about is the "@with_override" behavior over returning an empty dictionary. Seems wildly side-affecty. Regardless, I will try and see if all the tests pass

Reply:
👍 

> Seems wildly side-affecty

sparkmagic codebase in a nutshell 😂